src/main/java/com/github/f0xdx/Schemas.java

```diff
@@ -113,4 +119,101 @@ public static SchemaBuilder toBuilder(@NonNull Schema schema) {
 
     return builder;
   }
+
+  /**
+   * Returns a schema for storing the message headers
+   *
+   * @param headers
+   * @return
+   */
+  public static Schema forHeaders(Headers headers) {
+    final Map<String, List<Schema>> schemas = new HashMap<>();
+    StreamSupport.stream(headers.spliterator(), false)
+        .forEach(
+            header -> {
+              if (!schemas.containsKey(header.key())) {
+                schemas.put(header.key(), new ArrayList<>());
+              }
+              schemas.get(header.key()).add(header.schema());
+            });
+    final SchemaBuilder builder = SchemaBuilder.struct();
+    schemas.forEach((key, value) -> builder.field(key, mergeSchemas(value)));
+    return builder.build();
+  }
+
+  private static Schema mergeSchemas(List<Schema> headers) {
+    Schema schema = null;
+    for (Schema headerSchema : headers) {
+      schema = mergeSchemas(schema, headerSchema);
+    }
+    return SchemaBuilder.array(schema).optional().build();
+  }
+
+  private static Schema mergeSchemas(Schema schema, Schema addition) {
+    if (schema == null) {
+      return addition;
+    }
+    switch (addition.type()) {
+      case FLOAT32:
+      case FLOAT64:
+        if (schema.type().equals(Schema.Type.FLOAT32)
```

Comment:
It's always a good idea to invert this type of checks against constants - this rules out the unlikely case of a `null` type ruining your day :). Thanks to type being an enum, however, we can go with:

```suggestion
        if (schema.type() == Schema.Type.FLOAT32)
```